### ISSUE TITLE
Logs are not displaying while publishing the catalog to GitHub

### ISSUE THREAD

**Issue by mysticmode** [backstage/backstage#6546]
I'm trying to setup Backstage on Ubuntu 20.04 VPS. My problem is that when I try to "Choose" a template and publish it to GitHub, the logs are not getting displayed. I have tried this in both simply `yarn dev` or setting up a multistage `Dockerfile` and running it.

NOTE: The repository is getting published to GitHub fine. In other words, I can see the template that I want to deploy to GitHub fine. It is just that the `/create/task/<task-id>` says "Waiting for logs".

When I opened the Firefox console, it says this error:
`Firefox can’t establish a connection to the server at https://magna5.mysticmo.de/api/scaffolder/v2/tasks/f78f941e-3e05-4967-93b1-77dbd7034b7f/eventstream.`

Docker is installed -> Docker version 20.10.7

## Expected Behavior

The logs should get displayed.

## Current Behavior

Repository is getting deployed on GitHub, logs for that are not working.

<img width="1788" alt="image" src="https://user-images.githubusercontent.com/1125199/126271041-041edbf7-766b-43e2-abe6-8d2ad1d5823d.png">

## Steps to Reproduce

1. Create new app using `npx @backstage/create-app`
2. Write the app-config.local.yaml configuration (shown below)
3. `yarn install && yarn tsc && yarn build`
4. `yarn dev`

## Context

Template - https://github.com/magna5-test/golang-rest-template

1. app-config.yaml
```
app:
  title: Scaffolded Backstage App
  baseUrl: http://localhost:3000

organization:
  name: My Company

backend:
  baseUrl: http://localhost:7000
  listen:
    port: 7000
  csp:
    connect-src: ["'self'", 'http:', 'https:']
    # Content-Security-Policy directives follow the Helmet format: https://helmetjs.github.io/#reference
    # Default Helmet Content-Security-Policy values can be removed by setting the key to false
  cors:
    origin: http://localhost:3000
    methods: [GET, POST, PUT, DELETE]
    credentials: true
  # config options: https://node-postgres.com/api/client
  database:
    client: pg
    connection:
      host: ${POSTGRES_HOST}
      port: ${POSTGRES_PORT}
      user: ${POSTGRES_USER}
      password: ${POSTGRES_PASSWORD}
      # https://node-postgres.com/features/ssl
      #ssl: require # see https://www.postgresql.org/docs/current/libpq-ssl.html Table 33.1. SSL Mode Descriptions (e.g. require)
        #ca: # if you have a CA file and want to verify it you can uncomment this section
        #  $file: <file-path>/ca/server.crt
  cache:
    store: memory
  # workingDirectory: /tmp # Use this to configure a working directory for the scaffolder, defaults to the OS temp-dir

integrations:
  github:
    - host: github.com
      token: ${GITHUB_TOKEN}
    ### Example for how to add your GitHub Enterprise instance using the API:
    # - host: ghe.example.net
    #   apiBaseUrl: https://ghe.example.net/api/v3
    #   token: ${GHE_TOKEN}

proxy:
  '/test':
    target: 'https://example.com'
    changeOrigin: true

# Reference documentation http://backstage.io/docs/features/techdocs/configuration
# Note: After experimenting with basic setup, use CI/CD to generate docs
# and an external cloud storage when deploying TechDocs for production use-case.
# https://backstage.io/docs/features/techdocs/how-to-guides#how-to-migrate-from-techdocs-basic-to-recommended-deployment-approach
techdocs:
  builder: 'local' # Alternatives - 'external'
  generators:
    techdocs: 'docker' # Alternatives - 'local'
  publisher:
    type: 'local' # Alternatives - 'googleGcs' or 'awsS3'. Read documentation for using alternatives.

auth:
  # see https://backstage.io/docs/auth/ to learn about auth providers
  providers: {}

scaffolder:
  github:
    token: ${GITHUB_TOKEN}
    visibility: public # or 'internal' or 'private'

catalog:
  rules:
    - allow: [Component, System, API, Group, User, Resource, Location]
  locations:
    ...
```

2. app-config.local.yaml
```
app:
  title: Magna5 Backstage portal
  baseUrl: https://magna5.mysticmo.de

backend:
  baseUrl: https://magna5.mysticmo.de
  client: pg
  connection:
    host:
      $env: POSTGRES_HOST
    port:
      $env: POSTGRES_PORT
    user:
      $env: POSTGRES_USER
    password: 
      $env: POSTGRES_PASSWORD
    database:
      $env: POSTGRES_DATABASE

organization:
  name: Magna5 Backstage portal

auth:
  providers:
    github:
      development:
        clientId:
          $env: AUTH_GITHUB_CLIENT_ID
        clientSecret:
          $env: AUTH_GITHUB_CLIENT_SECRET

integrations:
  github:
    - host: github.com
      token:
        $env: GITHUB_TOKEN

scaffolder:
  defaultAuthor:
    name: Nirmal Almara
    email: hello@nlmara.com
  github:
    token:
      $env: GITHUB_TOKEN


techdocs:
  builder: 'local'
  publisher:
    type: 'local'
  generators:
    techdocs: local

catalog:
  processors:
    githubOrg:
      providers:
        - target: https://github.com
          token:
            $env: GITHUB_TOKEN
  rules:
    - allow: [Component, System, API, Group, User, Resource, Location]
  locations:
    - type: url
       target: https://github.com/magna5-test/golang-rest-template/blob/master/template.yaml
```

## Your Environment

- NodeJS Version (v12): v14.17.2
- Operating System and Version (e.g. Ubuntu 14.04): Ubuntu 20.04
- Browser Information: Firefox 89.0.2

**Comment by benjdlambert** [comment]
Hey @mysticmode how is this deployed? Is it just on a machine or k8s? Also what do you have in front of it? Load balancer or Ingress, nginx? or are you just hitting the machine directly?

**Comment by mysticmode** [comment]
Hi @benjdlambert - Thanks for the reply!

I'm doing with multi-stage Docker build and NGINX. This is the Dockerfile in the root directory [taken from Backstage Docs](https://backstage.io/docs/deployment/docker#multi-stage-build) with little modifications where I have added `cookiecutter` and couple other packages via `pip`. On `CMD` - `app-config.local.yaml` is included as well.
```Dockerfile
# Stage 1 - Create yarn install skeleton layer
FROM node:14-buster-slim AS packages

WORKDIR /app
COPY package.json yarn.lock tsconfig.json ./

COPY packages packages
# COPY plugins plugins

RUN find packages \! -name "package.json" -mindepth 2 -maxdepth 2 -exec rm -rf {} \+

# Stage 2 - Install dependencies and build packages
FROM node:14-buster-slim AS build

WORKDIR /app
COPY --from=packages /app .

RUN yarn install --frozen-lockfile --network-timeout 600000 && rm -rf "$(yarn cache dir)"

COPY . .

RUN yarn tsc
RUN yarn --cwd packages/backend backstage-cli backend:bundle --build-dependencies

# Stage 3 - Build the actual backend image and install production dependencies
FROM node:14-buster-slim

WORKDIR /app

# Copy the install dependencies from the build stage and context
COPY --from=build /app/yarn.lock /app/package.json /app/tsconfig.json /app/packages/backend/dist/skeleton.tar.gz ./
RUN tar xzf skeleton.tar.gz && rm skeleton.tar.gz

RUN yarn install --frozen-lockfile --production --network-timeout 600000 && rm -rf "$(yarn cache dir)"

# Copy the built packages from the build stage
COPY --from=build /app/packages/backend/dist/bundle.tar.gz .
RUN tar xzf bundle.tar.gz && rm bundle.tar.gz

# Copy any other files that we need at runtime
COPY app-config.yaml app-config.local.yaml ./

RUN apt-get update && apt-get install -y python3 python3-pip
RUN pip3 install mkdocs-techdocs-core==0.0.16 cookiecutter==1.7.2 jinja2_custom_filters_extension==0.0.2

CMD ["node", "packages/backend", "--config", "app-config.yaml", "--config", "app-config.local.yaml"]
```
I'm running it with the below commands in the project root directory.
```
yarn install && yarn tsc && yarn build
docker run -d --network=host --env-file ./.env.list -it -p 7000:7000 backstage
 ```
You could check the Nginx config below:
```Nginx
server {
    listen 80;
    listen [::]:80;
    server_name magna5.mysticmo.de;
    return 301 https://magna5.mysticmo.de$request_uri;
}

server {
    listen [::]:443 ssl ipv6only=on; # managed by Certbot
    listen 443 ssl; # managed by Certbot
    server_name magna5.mysticmo.de;

    ssl_certificate /etc/letsencrypt/live/magna5.mysticmo.de/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/magna5.mysticmo.de/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot
    
    location / {
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $remote_addr;
        proxy_set_header Host $host;
        proxy_pass http://127.0.0.1:7000;
    }
}
```

**Comment by benjdlambert** [comment]
@mysticmode, so I think this as something to do with nginx config and the location config not really supporting the connection of event sources. Can you try following this and report back?

https://stackoverflow.com/questions/13672743/eventsource-server-sent-events-through-nginx

**Comment by mysticmode** [comment]
You are correct, the settings for NGINX is not correct for event sources to work. The stackoverflow link you've shared helped me correct the NGINX script. Now I have checked, the logs are getting displayed correctly.

For future references, I'm sharing the NGINX config script here:
```nginx
server {
    listen 80;
    listen [::]:80;
    server_name magna5.mysticmo.de;
    return 301 https://magna5.mysticmo.de$request_uri;
}

server {
    listen [::]:443 ssl ipv6only=on; # managed by Certbot
    listen 443 ssl; # managed by Certbot
    server_name magna5.mysticmo.de;

    ssl_certificate /etc/letsencrypt/live/magna5.mysticmo.de/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/magna5.mysticmo.de/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot
    
    location / {
        proxy_set_header Connection '';
	proxy_http_version 1.1;
	chunked_transfer_encoding off;
	proxy_buffering off;
	proxy_cache off;

        proxy_set_header Host $host;
        proxy_pass http://127.0.0.1:7000;
    }
}
```

@benjaminr - Thank you!

**Comment by benjdlambert** [comment]
Ok perfect - I'm wondering if we should add this to documentation somewhere too. Do you think you could find a suitable place as to where it could go seen as you've just been through this and probably have a good idea where it should be? :)

**Comment by mysticmode** [comment]
IMO - it helps definitely! We could refer some other self-hosted applications. The first place I looked for is Deployment when I did this.

[Gitea comes to my mind now](https://docs.gitea.io/en-us/reverse-proxies/).

We could put up a separate section under `Deployment` saying "**Setting up reverse proxies**" and we could add NGINX, Apache, Caddy etc,. with appropriate configurations.

I'm going to ping you on Discord and see if I can be of any help :)
